typegen/src/utils.rs

```diff
@@ -330,6 +384,380 @@ mod tests {
         TypeParameter,
     };
 
+    #[test]
+    fn assoc_generics() {
```

Comment:
So this is checking that associated types lead to deduplication?

I'd probable tweak the name to something like `associated_types_are_properly_deduplicated` or add a comment to make it clearer what it's testing :)

Reply:
renamed to `associated_types_are_properly_deduplicated`